sycl/source/detail/graph_impl.cpp

```diff
@@ -596,11 +596,40 @@ exec_graph_impl::enqueue(const std::shared_ptr<sycl::detail::queue_impl> &Queue,
   sycl::detail::EventImplPtr NewEvent;
 
   if (CommandBuffer) {
-    if (!previousSubmissionCompleted()) {
-      throw sycl::exception(make_error_code(errc::invalid),
-                            "This Graph cannot be submitted at the moment "
-                            "because the previous run has not yet completed.");
+    // if previous submissions are incompleted, we automatically
+    // add completion events of previous submissions as dependencies.
+    // With Level-Zero backend we cannot resubmit a command-buffer until the
+    // previous one has already completed.
+    // Indeed, since a command-list does not accept a list a dependencies at
+    // submission, we circumvent this lack by adding a barrier that waits on a
+    // specific event and then define the conditions to signal this event in
+    // another command-list. Consequently, if a second submission is performed,
+    // the signal conditions of this single event are redefined by this second
+    // submission. Thus, this can lead to an undefined behaviour and potential
+    // hangs. We have therefore to expliclty wait in the host for
+    // previous submission to complete before resubmitting the command-buffer
+    // for level-zero backend.
+    // TODO : add a test to release this constraint and allow mulitple
```

Comment:
"test" makes me think of E2E and unittests 
```suggestion
    // TODO : add a check to release this constraint and allow multiple
```